x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/DataTypeConverter.java

```diff
@@ -95,6 +96,10 @@ public static DataType commonType(DataType left, DataType right) {
             }
         }
 
+        if (isDateTime(left) && isDateTime(right)) {
+            return DATETIME;
```

Comment:
Correct, this is what we want. DATETIME_NANOS is used actually only for the field extractors where currently the DATETIME is returned as long millis and the DATETIME_NANOS as a complete string. This will be eliminated once this functionality is merged.